irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -100,8 +104,9 @@ OnDemandOrderingGate::processProposalRequest(
 void OnDemandOrderingGate::sendCachedTransactions() {
   // TODO mboldyrev 22.03.2019 IR-425
   // make cache_->getBatchesForRound(current_round) that respects sync
-  auto batches = cache_->pop();
-  cache_->addToBack(batches);
+
+  cache_->rotate();
+  auto batches = cache_->back();
```

Comment:
It will be better to call `auto batches = front(); cache_->rotate();` to prevent new batches being accidentally resent due to concurrency.